console2/src/components/organisms/RestartProcessPopup/index.tsx

```diff
@@ -0,0 +1,91 @@
+/*-
+ * *****
+ * Concord
+ * -----
+ * Copyright (C) 2017 - 2018 Walmart Inc.
+ * -----
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import * as React from 'react';
+
+import { ConcordId, RequestError } from '../../../api/common';
+import { SingleOperationPopup } from '../../molecules';
+import { memo, useCallback } from 'react';
+import { restart as apiRestart } from '../../../api/process';
+import { useState } from 'react';
+import {Message} from "semantic-ui-react";
+
+interface ExternalProps {
+    instanceId: ConcordId;
+    parentInstanceId?: ConcordId;
+    refresh: () => void;
+    trigger: (onClick: () => void) => React.ReactNode;
+}
+
+const RestartProcessPopup = memo((props: ExternalProps) => {
+    const [restarting, setRestarting] = useState(false);
+    const [error, setError] = useState<RequestError>();
+    const [success, setSuccess] = useState(false);
+
+    const instanceId = props.instanceId;
+    const parentInstanceId = props.parentInstanceId;
+
+    const cancelProcess = useCallback(async () => {
+        setRestarting(true);
+
+        try {
+            await apiRestart(instanceId);
+            setSuccess(true);
+        } catch (e) {
+            setError(e);
+        } finally {
+            setRestarting(false);
+        }
+    }, [instanceId]);
+
+    const reset = useCallback(() => {
+        setRestarting(false);
+        setSuccess(false);
+        setError(undefined);
+    }, []);
+
+    const { trigger, refresh } = props;
+
+    return (
+        <SingleOperationPopup
+            trigger={trigger}
+            title="Restart the process?"
+            introMsg={
+            <>
+                <p>Are you sure you want to restart the selected process?</p>
```

Comment:
Something like this, maybe.

```suggestion
{!parentInstanceId && <p>Are you sure you want to restart the selected process?</p>}
{parentInstanceId &&
<Message warning>
You are about to restart a child process. Please note:

    Only the <a href="...">parent (root) process</a> will be restarted.
    Restarting the parent process may also re-run this and other child processes.
    Ensure that this is the desired action before proceeding.

Do you want to continue with restarting the parent process?
</Message>}
```